testing/matrix-sdk-integration-testing/src/tests/e2ee.rs

```diff
@@ -784,3 +790,193 @@ async fn test_cross_signing_bootstrap() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn test_secret_gossip_after_interactive_verification() -> Result<()> {
+    let encryption_settings = EncryptionSettings {
+        auto_enable_cross_signing: true,
+        auto_enable_backups: true,
+        backup_download_strategy: BackupDownloadStrategy::OneShot,
+    };
+
+    let first_client = SyncTokenAwareClient::new(
+        TestClientBuilder::new("alice_gossip_test")
+            .randomize_username()
+            .encryption_settings(encryption_settings)
+            .build()
+            .await?,
+    );
+
+    let user_id = first_client.user_id().expect("We should have access to the user id now");
+
+    let request = CreateRoomRequest::new();
+    let room_first_client = first_client.create_room(request).await?;
+    room_first_client.enable_encryption().await?;
+    first_client.sync_once().await?;
+
+    first_client.encryption().recovery().enable().await?;
+
+    assert_eq!(first_client.encryption().recovery().state(), RecoveryState::Enabled);
+
+    let response = room_first_client
+        .send(RoomMessageEventContent::text_plain("It's a secret to everybody"))
+        .await?;
+
+    let event_id = response.event_id;
+
+    warn!("The first device has created and enabled encryption in the room and sent an event");
+
+    let second_client = SyncTokenAwareClient::new(
+        TestClientBuilder::new(user_id.localpart())
+            .encryption_settings(encryption_settings)
+            .build()
+            .await?,
+    );
+
+    second_client.encryption().wait_for_e2ee_initialization_tasks().await;
+
+    // The second client doesn't have access to the backup, nor is recovery in the
+    // enabled state.
+    assert_eq!(second_client.encryption().recovery().state(), RecoveryState::Incomplete);
+    assert_eq!(second_client.encryption().backups().state(), BackupState::Unknown);
+
+    warn!("The first device: {}", first_client.device_id().unwrap());
+    warn!("The second device: {}", second_client.device_id().unwrap());
+    assert_ne!(first_client.device_id().unwrap(), second_client.device_id().unwrap());
+
+    second_client.sync_once().await?;
+
+    let first_device = second_client
```

Comment:
Should we call this something different to make it obvious it's the second client's one? `seconds_first_device`?